packages/cohttp/cohttp.3.0.0/opam

```diff
@@ -0,0 +1,67 @@
+opam-version: "2.0"
+maintainer: "anil@recoil.org"
+authors: [
+  "Anil Madhavapeddy"
+  "Stefano Zacchiroli"
+  "David Sheets"
+  "Thomas Gazagnaire"
+  "David Scott"
+  "Rudi Grinberg"
+  "Andy Ray"
+]
+synopsis: "An OCaml library for HTTP clients and servers"
+description: """
+Cohttp is an OCaml library for creating HTTP daemons. It has a portable
+HTTP parser, and implementations using various asynchronous programming
+libraries.
+
+See the cohttp-async, cohttp-lwt, cohttp-lwt-unix, cohttp-lwt-jsoo and
+cohttp-mirage libraries for concrete implementations for particular
+targets.
+
+You can implement other targets using the parser very easily. Look at the `IO`
+signature in `lib/s.mli` and implement that in the desired backend.
+
+You can activate some runtime debugging by setting `COHTTP_DEBUG` to any
+value, and all requests and responses will be written to stderr.  Further
+debugging of the connection layer can be obtained by setting `CONDUIT_DEBUG`
+to any value."""
+license: "ISC"
+tags: ["org:mirage" "org:xapi-project"]
+homepage: "https://github.com/mirage/ocaml-cohttp"
+doc: "https://mirage.github.io/ocaml-cohttp/"
+bug-reports: "https://github.com/mirage/ocaml-cohttp/issues"
+depends: [
+  "ocaml" {>= "4.04.1"}
```

Comment:
```suggestion
  "ocaml" {>= "4.08"}
```
Since we use `Bool` and `Int` and other "new" modules, we need to require `4.08` now.
@hannesm @dinosaure  should we use stdcompat and restore backwards compatibility to older versions of ocaml or you are also fine with dropping the ball to `< 4.08`?

Reply:
I answer myself. It will not matter since all other packages have similar issues:
```
# Detecting depexts using vars: arch=x86_64, os=linux, os-distribution=debian, os-family=debian
[ERROR] No solution for cohttp-lwt-unix.3.0.0: The following dependencies couldn't be met:
          - cohttp-lwt-unix -> conduit-lwt-tls -> mirage-crypto-rng >= 0.8.0 -> ocaml >= 4.08.0
              base of this switch (use `--unlock-base' to force)
```
and similarly for async.

Reply:
I can still make it backward compatible for some versions with`cohttp-lwt` if `cohttp-mirage` needs to support compilers older than `4.08`, just let me know.

Reply:
fine with me

Reply:
Fine with me too 👍

Reply:
Fantastic, let's see how it goes now then